source/common/http/http1/balsa_parser.cc

```diff
@@ -143,7 +143,12 @@ ParserStatus BalsaParser::getStatus() const { return status_; }
 uint16_t BalsaParser::statusCode() const { return headers_.parsed_response_code(); }
 
 bool BalsaParser::isHttp11() const {
-  return absl::EndsWith(headers_.first_line(), Http::Headers::get().ProtocolStrings.Http11String);
+  if (framer_.is_request()) {
+    return absl::EndsWith(headers_.first_line(), Http::Headers::get().ProtocolStrings.Http11String);
+  } else {
+    return absl::StartsWith(headers_.first_line(),
```

Comment:
Is it worth adding a unit test that exercises this case?

Reply:
Good question.

Four existing tests fail if I change the request case to `return true`.  18 existing tests (plus two new expectations) fail if I change the request case to `return false`.  Many of the failures are at https://github.com/envoyproxy/envoy/blob/3b8beabfca22af97f9626ed399bfc14e88413137/test/common/http/http1/codec_impl_test.cc#L231.  It seems like both cases are sufficiently covered.